src/modules/keyboardmanager/ui/KeyDropDownControl.cpp

```diff
@@ -125,177 +125,208 @@ void KeyDropDownControl::SetSelectionHandler(Grid& table, StackPanel& singleKeyC
 {
-    auto onSelectionChange = [&, table, shortcutControl, colIndex, parent](winrt::Windows::Foundation::IInspectable const& sender) {
-        ComboBox currentDropDown = sender.as<ComboBox>();
-        int selectedKeyIndex = currentDropDown.SelectedIndex();
-        uint32_t dropDownIndex = -1;
-        bool dropDownFound = parent.Children().IndexOf(currentDropDown, dropDownIndex);
-        // Get row index of the single key control
-        uint32_t controlIndex;
-        bool controlIindexFound = table.Children().IndexOf(shortcutControl, controlIndex);
-        KeyboardManagerHelper::ErrorType errorType = KeyboardManagerHelper::ErrorType::NoError;
-        bool IsDeleteDropDownRequired = false;
+    ComboBox currentDropDown = dropDown;
+    int selectedKeyIndex = currentDropDown.SelectedIndex();
+    uint32_t dropDownIndex = -1;
+    bool dropDownFound = parent.Children().IndexOf(currentDropDown, dropDownIndex);
+    // Get row index of the single key control
+    uint32_t controlIndex;
+    bool controlIindexFound = table.Children().IndexOf(shortcutControl, controlIndex);
+    KeyboardManagerHelper::ErrorType errorType = KeyboardManagerHelper::ErrorType::NoError;
+    bool IsDeleteDropDownRequired = false;
+    int rowIndex = -1;
 
-        if (controlIindexFound)
+    if (controlIindexFound)
+    {
+        rowIndex = (controlIndex - KeyboardManagerConstants::ShortcutTableHeaderCount) / KeyboardManagerConstants::ShortcutTableColCount;
+        if (selectedKeyIndex != -1 && keyCodeList.size() > selectedKeyIndex && dropDownFound)
         {
-            int rowIndex = (controlIndex - KeyboardManagerConstants::ShortcutTableHeaderCount) / KeyboardManagerConstants::ShortcutTableColCount;
-            if (selectedKeyIndex != -1 && keyCodeList.size() > selectedKeyIndex && dropDownFound)
+            // If only 1 drop down and action key is chosen: Warn that a modifier must be chosen
+            if (parent.Children().Size() == 1 && !KeyboardManagerHelper::IsModifierKey(keyCodeList[selectedKeyIndex]))
             {
-                // If only 1 drop down and action key is chosen: Warn that a modifier must be chosen
-                if (parent.Children().Size() == 1 && !KeyboardManagerHelper::IsModifierKey(keyCodeList[selectedKeyIndex]))
-                {
-                    // warn and reset the drop down
-                    errorType = KeyboardManagerHelper::ErrorType::ShortcutStartWithModifier;
-                }
-                // If it is the last drop down
-                else if (dropDownIndex == parent.Children().Size() - 1)
+                // warn and reset the drop down
+                errorType = KeyboardManagerHelper::ErrorType::ShortcutStartWithModifier;
+            }
+            // If it is the last drop down
+            else if (dropDownIndex == parent.Children().Size() - 1)
+            {
+                // If last drop down and a modifier is selected: add a new drop down (max of 5 drop downs should be enforced)
```

Comment:
nit: max of 3 drop downs should be enforced?

Reply:
Ill remove the mention of count, since its already stored as a constant. I'll change it to `max drop down count should be enforced`. That way if we increase it in the future it doesn't need to be changed.